sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusSessionProcessor.cs

```diff
@@ -19,7 +22,7 @@ namespace Azure.Messaging.ServiceBus
     /// property. The error handler is specified with the <see cref="ProcessErrorAsync"/> property.
     /// To start processing after the handlers have been specified, call <see cref="StartProcessingAsync"/>.
     /// </summary>
-    public class ServiceBusSessionProcessor
+    public class ServiceBusSessionProcessor : IAsyncDisposable
```

Comment:
Is it a breaking change to make a type go from non-disposable to disposable? Mainly, programs that were using this prior would not be calling `.Dispose()`. Thus, the 'usage contract' has changed (if that makes sense).

Reply:
I think as long as you make sure that old code could work correctly without calling Dispose, it wouldn't necessarily be breaking. But this is a preview so it should be okay 😄 

https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#types